saleor/graphql/product/resolvers.py

```diff
@@ -141,9 +142,7 @@ def resolve_products(
     **_kwargs,
 ):
 
-    user = info.context.user
```

Comment:
Probably we should use `get_requester_from_context` in almost all places where `user = info.context.user` appear in code. All query and mutation should work for user and service account with proper permissions.

Reply:
Sometimes we want to get only user and sometimes user or service account. This is the second case. For getting user from the context we could also have a helper.